tools/swift-format/Sources/Rules/MaximumBlankLines.swift

```diff
@@ -16,5 +16,69 @@ import SwiftSyntax
 ///
 /// - SeeAlso: https://google.github.io/swift#vertical-whitespace
 public final class MaximumBlankLines: SyntaxFormatRule {
+  public override func visit(_ token: TokenSyntax) -> Syntax {
+    guard let parentToken = token.parent else {
+      return token.withLeadingTrivia(removeExtraBlankLines(token).0)
+    }
 
+    guard let grandparentTok = parentToken.parent else {
+      return token.withLeadingTrivia(removeExtraBlankLines(token).0)
+    }
+
+    // Tokens who appeared in a member type are handle by
+    // BlankLineBetweenMembers rule.
+    if grandparentTok is MemberDeclListItemSyntax {
+      return token
+    }
+    else {
+      let correctTrivia = removeExtraBlankLines(token)
+      return correctTrivia.1 ? token : token.withLeadingTrivia(removeExtraBlankLines(token).0)
+    }
+  }
+
+  /// Indicates if the given trivia contains an invalid amount of
+  /// consecutively blank lines. If it does it returns a clean trivia
+  /// with the correct amount of blank lines.
+  func removeExtraBlankLines(_ token: TokenSyntax) -> (Trivia, Bool) {
+    let maxBlankLines = context.configuration.maximumBlankLines
+    var pieces = [TriviaPiece]()
+    let isTheFirstOne = token == token.root.firstToken
+    let tokenTrivia = token.leadingTrivia
+    var startsIn = 0
+    var hasValidAmountOfBlankLines = true
+    let triviaWithoutTrailingSpaces = tokenTrivia.withoutTrailingSpaces()
+
+    // Ensures the beginning of file doesn't have an invalid amount of blank line.
+    // The first triviapiece of a file is a special case, where each newline is
+    // a blank line.
+    if isTheFirstOne, let firstPiece = triviaWithoutTrailingSpaces.first,
+       case TriviaPiece.newlines(let num) = firstPiece, num > maxBlankLines {
```

Comment:
You should be able to remove `TriviaPiece` here (and in a couple other places below). Our style guide prefers using type inference over explicitly stating the type.